editor/src/components/custom-code/code-file.ts

```diff
@@ -254,3 +273,169 @@ export function codeCacheToBuildResult(cache: {
     }
   }, cache)
 }
+
+export interface NormalisePathSuccess {
+  type: 'NORMALISE_PATH_SUCCESS'
+  normalisedPath: StaticInstancePath
+  filePath: string
+  textFile: TextFile
+}
+
+export function normalisePathSuccess(
+  normalisedPath: StaticInstancePath,
+  filePath: string,
+  textFile: TextFile,
+): NormalisePathSuccess {
+  return {
+    type: 'NORMALISE_PATH_SUCCESS',
+    normalisedPath: normalisedPath,
+    filePath: filePath,
+    textFile: textFile,
+  }
+}
+
+export interface NormalisePathEndsAtDependency {
+  type: 'NORMALISE_PATH_ENDS_AT_DEPENDENCY'
+  dependency: string
+}
+
+export function normalisePathEndsAtDependency(dependency: string): NormalisePathEndsAtDependency {
+  return {
+    type: 'NORMALISE_PATH_ENDS_AT_DEPENDENCY',
+    dependency: dependency,
+  }
+}
+
+export interface NormalisePathError {
+  type: 'NORMALISE_PATH_ERROR'
+  errorMessage: string
+}
+
+export function normalisePathError(errorMessage: string): NormalisePathError {
+  return {
+    type: 'NORMALISE_PATH_ERROR',
+    errorMessage: errorMessage,
+  }
+}
+
+export interface NormalisePathUnableToProceed {
```

Comment:
Seems like this could do with coming with an error message too?

Reply:
Could probably do with specifying the file that it's halted at so that an error could be constructed from it.

Reply:
Fixed in 9bf80891.